plugins/admin-settings/email-types/index.jsx

```diff
@@ -0,0 +1,164 @@
+import { useEffect, useState } from 'react';
+import { __ } from '@wordpress/i18n';
+import {
+  Button, SelectControl, TextControl, withNotices,
+} from '@wordpress/components';
+import ImagePicker from '@/components/imagePicker';
+import {
+  Checkboxes, Sortable, SortableItem, useOption,
+} from '@alleyinteractive/block-editor-tools';
+import apiFetch from '@wordpress/api-fetch';
+import { v4 as uuidv4 } from 'uuid';
+
+const AdminEmailTypes = withNotices(
+  ({ noticeOperations, noticeUI }) => {
+    const {
+      value: emailTypes = {},
+      isEdited = false,
+      isSaving = false,
+      onChange = () => {},
+      onSave = () => {},
+    } = useOption('nb_email_types');
+
+    const {
+      value: settings = {},
+    } = useOption('nb_settings');
+
+    const {
+      from_names: fromNames = [],
+    } = settings;
+
+    const fromNameOptions = fromNames.map(
+      (name) => ({ label: name, value: name }),
+    );
+
+    // Get all posts with the nb_template post type.
+    const [templatePosts, setTemplatePosts] = useState([]);
+    useEffect(() => {
+      apiFetch({ path: '/wp/v2/nb_template' })
```

Comment:
What happens if there are more than `$posts_per_page` templates? At a minimum we should fetch 100 per page, and it would be good to check the total number of pages and continue to fetch until we've got them all.